float8_experimental/float8_tensor.py

```diff
@@ -12,6 +13,30 @@
 aten = torch.ops.aten
 
 
+@dataclass(frozen=True)
+class ScaledMMConfig:
```

Comment:
does compile work if we just pass the booleans around without this data structure?

Reply:
Mi need to rebase and try again. probably, but I was frustrated because it feels like it should and was annoyed that compile made me have worse code quality, imo.

Cc @bdhirsh

Reply:
3 booleans is fine IMO, one could argue having a config for this might be overkill

Reply:
It's 3 bools today but the float dtype could also live in here which is needed for the AMD PR. I think this is just a stylistic thing and it is my preference to wrap things into structs when they are semantically related.

Reply:
if it works the naive way now, maybe we could land that and refactor later ? :) there is no BC right now

Reply:
https://github.com/openai/triton/issues/2513 shows pretty large speedups from this, so would be nice to make it easier to experiment with it

Reply:
> does compile work if we just pass the booleans around without this data structure?

Yeah - when you run `torch.compile`, we will try to either lower the `autograd.Function` here as a Higher Order Op in the graph, or if you're using `allow_in_graph`, dynamo will try to dump the call to this `autograd.Function` directly in the graph (to be traced by AOTAutograd later).

The problem is that for every function call in the FX graph, we are currently limited by the fact that we need every input to be a primitive FX/torchscript type (and not an arbitrary python type).

It seems like it would be nice to have a way to either:
(1) support putting a custom python type (that we know desugars into primitive types) into the graph
(2) have dynamo automatically do the "translation" for you, by generating a new autograd.Function that takes in the desugared version of your struct.

Either of those options is definitely work though